src/UI/toolbar/BatteryIndicator.qml

```diff
@@ -278,6 +322,117 @@ Item {
                         }
                     }
                 }
+                
+            }
+
+            SettingsGroupLayout {
+                heading:        qsTr("Battery State Display")
+                Layout.fillWidth: true
+                spacing: ScreenTools.defaultFontPixelWidth * 0.5
+                
+                RowLayout {
+                    //Layout.fillWidth: true
+                    
+                    QGCColoredImage {
+                        source: "/qmlimages/BatteryGreen.svg"
+                        height: ScreenTools.defaultFontPixelHeight * 5
+                        width: ScreenTools.defaultFontPixelWidth * 7
+                        fillMode: Image.PreserveAspectFit
+                        color: qgcPal.colorGreen
+                    }
+
+                    QGCLabel {
+                        Layout.preferredWidth: ScreenTools.defaultFontPixelWidth * 3.5 
+                        text: qsTr("100%")
+                        Layout.leftMargin: -13  
+                    }
+
+                    QGCColoredImage {
+                        source: "/qmlimages/BatteryYellowGreen.svg"
+                        height: ScreenTools.defaultFontPixelHeight * 5
+                        width: ScreenTools.defaultFontPixelWidth * 7
+                        fillMode: Image.PreserveAspectFit
+                        color: qgcPal.colorYellowGreen
+                    }
+
+                    FactTextField {
+                        id: threshold1Field
+                        fact: batterySettings.threshold1
+                        validator: IntValidator { bottom: 16 } // Value is at least 16
+                        implicitWidth: ScreenTools.defaultFontPixelWidth * 6  // Reduced width
+                        Layout.leftMargin: -10
+                        //Layout.preferredWidth: ScreenTools.defaultFontPixelWidth * 2  // Reduced width using Layout.preferredWidth
+                        height: ScreenTools.defaultFontPixelHeight * 1.5
+                        onEditingFinished: {
```

Comment:
Then you do final level of validation on a rawValueChanged signal to adjust if needed.

Reply:
I've implemented the final level of validation in the `BatteryIndicatorSettings.cc`. For `threshold1`, I ensure that the value is always at least 16 and no more than 99. If the provided value is below 16, it gets adjusted to 17. If it exceeds 99, it's capped at 99. Additionally, if the new value is less than or equal to `threshold2`, it automatically gets set to one more than `threshold2` to maintain the required order. For `threshold2`, I enforce a minimum value of 16. If the user attempts to set it to 15 or less, it gets adjusted up to 16. Moreover, if the new value is greater than or equal to `threshold1`, it is set to one less than `threshold1` to ensure it remains less.